src/test/java/ch/interlis/iox_j/validator/ViewConstraintsTest.java

```diff
@@ -0,0 +1,285 @@
+
+        AssertAllEventMessages(
+                logger.getErrs(),
+                "Set Constraint ViewConstraintsSet.TopicA.ViewSet.ObjectCount is not true."
+        );
+    }
+
+    @Test
+    public void extendingConstraintOk() {
+        LogCollector logger = validateObjects(
+                EXTENDING_TOPIC,
+                createTestObjectNumberAndText(CLASS_EXTENDING, "o1", 12, "Some value"),
+                createTestObjectNumberAndText(CLASS_EXTENDING, "o2", 250, "Another value"),
+                createTestObjectNumberAndText(CLASS_EXTENDING, "o3", 876, "Some other value")
+        );
+
+        assertEquals(0, logger.getErrs().size());
+    }
+
+    @Test
+    public void extendingConstraintFails() {
+        LogCollector logger = validateObjects(
+                EXTENDING_TOPIC,
+                createTestObjectNumberAndText(CLASS_EXTENDING, "o1", 12, "Some value"),
+                createTestObjectNumberAndText(CLASS_EXTENDING, "o2", 250, "A"),
+                createTestObjectNumberAndText(CLASS_EXTENDING, "o3", 150, "1234")
+        );
+
+        AssertAllEventMessages(
+                logger.getErrs(),
+                "Mandatory Constraint ViewConstraints.TopicA.ViewBase.LessThan100OrMoreThan200 is not true.",
+                "Mandatory Constraint ViewConstraintsExtending.ExtendingTopicA.ViewExtending.NewTextLength is not true."
+        );
+    }
+
+    private IomObject createTestObjectName(String className, String oid, String name) {
+        IomObject iomObj = new Iom_jObject(className, oid);
+        iomObj.setattrvalue("Name", name);
+        return iomObj;
+    }
+
+    private IomObject createTestObjectNumber(String className, String oid, double number) {
+        IomObject iomObj = new Iom_jObject(className, oid);
+        iomObj.setattrvalue("Number", Double.toString(number));
+        return iomObj;
+    }
+
+    private IomObject createTestObjectNumberAndText(String className, String oid, double number, String text) {
+        IomObject iomObj = createTestObjectNumber(className, oid, number);
+        iomObj.setattrvalue("Text", text);
+        return iomObj;
+    }
+
+    private LogCollector validateObjects(String topic, IomObject... objects) {
+        ValidationConfig modelConfig = new ValidationConfig();
+        LogCollector logger = new LogCollector();
+        validateObjects(topic, modelConfig, logger, objects);
+        return logger;
+    }
+
+    private LogCollector validateObjectsWithAdditionalModel(String topic, String models, IomObject... objects) {
+        ValidationConfig modelConfig = new ValidationConfig();
+        modelConfig.setConfigValue(ValidationConfig.PARAMETER, ValidationConfig.ADDITIONAL_MODELS, models);
+        LogCollector logger = new LogCollector();
+        validateObjects(topic, modelConfig, logger, objects);
+        return logger;
+    }
+
+    private void validateObjects(String topic, ValidationConfig modelConfig, LogCollector logger, IomObject... objects) {
```

Comment:
PP: Funktion extrahieren. Sie könnte allenfalls auch in anderen Tests gut wiederverwendet werden.